com.unity.shadergraph/Editor/Data/Graphs/Vector3MaterialSlot.cs

```diff
@@ -31,14 +42,22 @@ public Vector3MaterialSlot()
             SlotType slotType,
             Vector3 value,
             ShaderStageCapability stageCapability = ShaderStageCapability.All,
-            string label1 = "X",
-            string label2 = "Y",
-            string label3 = "Z",
+            string label1 = null,
+            string label2 = null,
+            string label3 = null,
             bool hidden = false)
             : base(slotId, displayName, shaderOutputName, slotType, stageCapability, hidden)
         {
             m_Value = value;
-            m_Labels = new[] { label1, label2, label3 };
+            if ((label1 != null) || (label2 != null) || (label3 != null))
```

Comment:
Is this condition intended to keep m_labels null if all of label1-3 are not, and if not all null then fill in the empty slots with defaults? The purpose of this is to avoid serializing the defaults?

Is there a specific reason why we don't want to serialize defaults? It could be more transparent and less maintenance to just serialize the defaults.

Reply:
Yes it is to keep from serializing default labels every time.
I dislike how it's serializing all of this redundant data that is 100% controlled by code.
But you might be right that it's simpler to just serialize them all the time, at least in the current setup.
I think the bigger issue is we don't have a clear separation of serialized state from our implementation.